stream/src/placeholder/instance.rs

```diff
@@ -63,18 +59,15 @@ impl<C: Scheme, Si: Sink, St: Stream> Instance<C, Si, St> {
             _ = runtime.sleep_until(deadline) => {
                 return Err(Error::HandshakeTimeout)
             },
-            result = stream.recv() => {
+            result = { network::read_next_frame(stream) } => {
                 result.map_err(|_| Error::ReadFailed)?
             },
         };
 
         // Verify handshake message from peer
         let handshake = Handshake::verify(
             runtime,
-            &config.crypto,
```

Comment:
nit: Let's avoid redoing the config-passing logic here...we can change this up in a separate PR if we still feel strongly about it.